Mage/src/main/java/mage/target/common/TargetCreaturePermanentAmount.java

```diff
@@ -11,20 +11,24 @@ public class TargetCreaturePermanentAmount extends TargetPermanentAmount {
 
     private static final FilterCreaturePermanent defaultFilter = new FilterCreaturePermanent("target creatures");
 
-    public TargetCreaturePermanentAmount(int amount) {
-        super(amount, defaultFilter);
+    public TargetCreaturePermanentAmount(int amount, int minNumberOfTargets) {
+        super(amount, minNumberOfTargets, defaultFilter);
+    }
+
+    public TargetCreaturePermanentAmount(int amount, int minNumberOfTargets, int maxNumberOfTargets) {
+        super(amount, minNumberOfTargets, maxNumberOfTargets, defaultFilter);
```

Comment:
I don't like how there are multiple inherited constructors calling super. Is there a reason you can't have a single constructor call super and the rest chain from it? (Or could be two, one for the int and one for the DynamicValue) Easier to follow imo.

Reply:
Good point. I've added two new big constructors and switched to chaining with them.